app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
   return element.nextSibling.children[0].dataset.explicit == 'true';
```

Comment:
The list I am going through is the list of serializable elements (in this case I am interested in the `select` element) and for each element I decide whether I should filter it out of the serialization.